public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -786,51 +785,86 @@ public SurveyQuestionResult apply(AchillesResult o) {
+        } catch(NullPointerException ie) {
+            throw new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");
+        }
         CdrVersion cdrVersion = cdrVersionDao.findByIsDefault(true);
+        if (cdrVersion == null) {
+            throw new ServerErrorException("No data present in table cdr version");
+        }
         return ResponseEntity.ok(TO_CLIENT_CDR_VERSION.apply(cdrVersion));
     }
 
     @Override
     public ResponseEntity<org.pmiops.workbench.model.Analysis> getGenderAnalysis(){
-        CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        try {
+            CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        } catch(NullPointerException ie) {
+            throw new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");
+        }
         AchillesAnalysis genderAnalysis = achillesAnalysisDao.findAnalysisById(GENDER_ANALYSIS);
+        if (genderAnalysis == null) {
+            throw new DataNotFoundException("Gender Analysis data is not available");
+        }
         addGenderStratum(genderAnalysis,1, "0", null);
         return ResponseEntity.ok(TO_CLIENT_ANALYSIS.apply(genderAnalysis));
     }
 
     @Override
     public ResponseEntity<org.pmiops.workbench.model.Analysis> getRaceAnalysis(){
-        CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        try {
+            CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        } catch(NullPointerException ie) {
+            throw new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");
+        }
         AchillesAnalysis raceAnalysis = achillesAnalysisDao.findAnalysisById(RACE_ANALYSIS);
+        if (raceAnalysis == null) {
+            throw new DataNotFoundException("Race Analysis data is not available");
+        }
         addRaceStratum(raceAnalysis);
         return ResponseEntity.ok(TO_CLIENT_ANALYSIS.apply(raceAnalysis));
     }
 
     @Override
     public ResponseEntity<org.pmiops.workbench.model.Analysis> getEthnicityAnalysis(){
-        CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        try {
+            CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        } catch(NullPointerException ie) {
+            throw new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");
+        }
         AchillesAnalysis ethnicityAnalysis = achillesAnalysisDao.findAnalysisById(ETHNICITY_ANALYSIS);
+        if (ethnicityAnalysis == null) {
+            throw new DataNotFoundException("Ethnicity Analysis data is not available");
+        }
```

Comment:
Can be
`Optional.ofNullable(ethnicityAnalysis).orElseThrow(() -> new DataNotFoundException("Ethnicity Analysis data is not available"));`

Reply:
Changed thanks.